cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -1415,22 +1435,69 @@ pub(crate) fn lower_icmp<C: LowerCtx<I = Inst>>(
                 }
             }
         }
+        condcode
     } else if !ty.is_vector() {
-        let alu_op = choose_32_64(ty, ALUOp::SubS32, ALUOp::SubS64);
         let rn = put_input_in_reg(ctx, inputs[0], narrow_mode);
         let rm = put_input_in_rse_imm12(ctx, inputs[1], narrow_mode);
+
+        let is_overflow = condcode == IntCC::Overflow || condcode == IntCC::NotOverflow;
+        let is_small_type = ty == I8 || ty == I16;
+        let (out_condcode, rn, rm) = if is_overflow && is_small_type {
+            // Overflow checks for non native types require additional instructions, other than
+            // just the extend op.
+            //
+            // TODO: Codegen improvements: Merge the second sxt{h,b} into the following sub instruction.
+            //
+            // sxt{h,b}  w0, w0
+            // sxt{h,b}  w1, w1
+            // sub       w0, w0, w1
+            // cmp       w0, w0, sxt{h,b}
+            //
+            // This comparision leaves the result in the `ne` flag, so we need to signal that
```

Comment:
`comparison`

Also, there is no `NE` flag - it is called `Z`. However, a better way to phrase this is to say `The result of this comparison is either the EQ or NE condition code ...`.